TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -1024,4 +1024,61 @@ public override int GetHashCode()
         }
     }
 
+    /// <summary>
+    /// Helper class for manipulating table definitions (DataDescription or DataRedefines having an OCCURS clause).
+    /// </summary>
+    public static class TableDefinitionExtensions
+    {
+        /// <summary>
+        /// Collect all parent table definitions of the given data-item.
+        /// If the given DataDefinition is a table itself, it is included in the result.
+        /// Results are ordered from the given data item to its highest parent table.
+        /// </summary>
+        /// <param name="dataDefinition">Data definition to start from.</param>
+        /// <returns>List of parent tables. Maybe null if the data-item is part of a TC typedef,
+        /// empty if the data-item does not belong to any table.</returns>
+        /// <remarks>Returns empty list for a null starting DataDefinition.</remarks>
+        public static List<DataDefinition> GetParentTableDefinitions(this DataDefinition dataDefinition)
```

Comment:
Well I hesitated on this one because the `null` return value only happens because of a lack in the parser (arrays in typedefs are not supported). But ok, we can put it on the method and remember to remove it when we add the support for arrays in typedefs.

Reply:
Done in e37c55a0a3d0e324126c197417a4c13597ec82ff.